client/my-sites/stats/stats-period-navigation/index.jsx

```diff
@@ -0,0 +1,39 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import Gridicon from 'components/gridicon';
+
+const StatsPeriodNavigation = ( { url, children, date, period, moment } ) => {
+	const isToday = moment( date ).isSame( moment(), period );
+	const previousDay = moment( date ).subtract( 1, period ).format( 'YYYY-MM-DD' );
```

Comment:
Not sure if we need to fix this or not, but the current logic allows for navigating to prior periods that are not shown in the chart

![stats_ _trout_bummin_ _wordpress_com](https://cloud.githubusercontent.com/assets/22080/21781167/2f84ef08-d663-11e6-97e1-49d0467d529b.png)

Granted you can do the same thing now by modifying the URL params, so guessing not a huge deal.